plugins/defaultplugins/ifplugin/data_resync.go

```diff
@@ -676,3 +695,189 @@ func (plugin *NatConfigurator) resolveMappings(nbDNatConfig *nat.Nat44DNat_DNatC
 		}
 	}
 }
+
+// Try to parse existing VPP interface with provided set of NB interfaces. If interfaces are the same type,
+// two parameters are compared:
+//	1. If both interfaces contain the same set of IP addresses, they are considered equal and will be correlated
+//  2. If first condition is false, compare MAC addresses; if both interfaces have the same MAC address,
+//     they will be also considered as correlated
+// Otherwise interfrace cannot be parsed and will be recreated
+func (plugin *InterfaceConfigurator) correlateInterfaces(nbIfs []*intf.Interfaces_Interface, vppIf *vppdump.Interface) *intf.Interfaces_Interface {
+	// find the best match
+	for nbIfIdx, nbIf := range nbIfs {
+		if nbIf.Type != vppIf.Type {
+			continue
+		}
+		// First attempt: match IP address
+		if len(nbIf.IpAddresses) == len(vppIf.IpAddresses) {
+			ipMatch := func(nbIPs, vppIPs []string) bool {
+				for _, nbIP := range nbIPs {
+					var ipFound bool
+					for _, vppIP := range vppIPs {
+						if nbIP == vppIP {
+							ipFound = true
+							break
+						}
+					}
+					if !ipFound {
+						return false
+					}
+				}
+				return true
+			}(nbIf.IpAddresses, vppIf.IpAddresses)
+			if ipMatch {
+				// Remove NB config (so agent knows it was already used)
+				nbIfs = append(nbIfs[:nbIfIdx], nbIfs[nbIfIdx+1:]...)
+				return nbIf
+			}
+		}
+		// Second attempt: match MAC
```

Comment:
Shouldn't the first attempt be by matching MAC?

Reply:
adding interface tags, a heuristic correlation was removed so this is out of date